MacBry-Maciej-Bryja/SimpleTodoApp1/src/main/java/com/mac/bry/simpleTodo/DAO/UserDAO.java

```diff
@@ -0,0 +1,198 @@
+package com.mac.bry.simpleTodo.DAO;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+import com.mac.bry.simpleTodo.DAO.API.DAOUserAPI;
+import com.mac.bry.simpleTodo.entity.Task;
+import com.mac.bry.simpleTodo.entity.User;
+
+public class UserDAO implements DAOUserAPI {
+	
+	private SessionFactory factory;
+	
+	public UserDAO() {
+		super();
+		this.factory = new Configuration()
+				.configure("hibernate.cfg.xml")
+				.addAnnotatedClass(User.class)
+				.addAnnotatedClass(Task.class)
+				.buildSessionFactory();
+	}
+	
+	@Override
+	public void addUser(User user) {
+		List<User> userList;
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		userList = session.createQuery("from User user where user.login='" + user.getLogin() + "'").getResultList();
+		if(userList.isEmpty()) {
+			session.save(user);
+			session.getTransaction().commit();
+			session.close();
+		}
+		else {
+			System.out.println("Incorrect login");	
+			session.getTransaction().commit();
+			session.close();
+		}
+		
+	}
+	
+	@Override
+	public void deleteUserByID(int id) {
+		User userToDelete = findUserByID(id);
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		session.delete(userToDelete);
+		session.getTransaction().commit();
+		session.close();
+	}
+	
+	@Override
+	public void deleteUserByLogin(String login) {
+		User userToDelete = findUserByLogin(login);
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		session.delete(userToDelete);
+		session.getTransaction().commit();
+	}
+
+	@Override
+	public void editUserLogin(User user, String newLogin) {
+		String oldLogin = user.getLogin();
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		user.setLogin(newLogin);
+		session.getTransaction().commit();
+		System.out.println("You have changed login from " + oldLogin + " to " + user.getLogin());
```

Comment:
Jak wyżej - logowanie przy użyciu sout. Jeśli chcesz wyświetlać informację o wykonanej akcji użytkownikowi to powinno dziać się to w kontrolerze.

Reply:
Zrobione za pomocą Log4j